src/app/modals/encryptwallet/encryptwallet.component.ts

```diff
@@ -36,29 +40,35 @@ export class EncryptwalletComponent implements OnInit {
 
         this.log.d(`Encrypting wallet! password: ${this.password}`);
         this._rpc.call('encryptwallet', [password.password])
-            .subscribe(
-            (response: any) => {
+          .subscribe(
+            response => {
+              this._rpc.toggleState(false);
               this.alertBox.open(response, 'Wallet');
-            } ,
+              setTimeout(() => {
+                this._rpc.call('restart-daemon')
+                  .subscribe(() => {
+                    this._modalsService.open('createWallet', {forceOpen: true});
```

Comment:
So how is the initial workflow?
We force pop up on the encrypt modal, and when it's encrypted and restarted we open the createWallet modal?

Reply:
Ahh I see, this one should be conditional, and the createWallet should only open when the encrypt modal was triggered by createwallet

Reply:
fix #148